index.rst

```diff
@@ -39,4 +39,261 @@
 
 :tocdepth: 1
 
-Write the docs; right here.
+Introduction
+============
+
+Python 2 will `cease to be supported in 2020 <https://www.python.org/dev/peps/pep-0373/>`_ and there is now `a push from the developer community <https://python3statement.github.io>`_ to drop support for Python 2 following upcoming long term support releases.
+This includes Jupyter, Matplotlib and `Astropy <https://github.com/astropy/astropy-APEs/blob/master/APE10.rst>`_.
+The `scientific community are also starting to investigate Python 3 features <http://python-3-for-scientists.readthedocs.io>`_ that may help them.
+Furthermore, the LSST commissioning and operations timeline indicates that we should be planning to use Python 3 in the deployed system.
+
+In August 2015 it was decided with `RFC-60 <https://jira.lsstcorp.org/browse/RFC-60>`_ that we should migrate the stack code (everything in ``lsst_distrib``) to support both Python 3 and Python 2.
+Packages outside of ``lsst_distrib`` have to support Python 3 but have the option of dropping Python 2 by filing an RFC demonstrating that they have no external users that will be affected.
+This document provides instructions on setting up a Python 3 environment and provides guidance on how to modify code to support both Python 2 and Python 3.
+
+A `preliminary guide on porting to Python 3 <http://dx.doi.org/10.5281/zenodo.60566>`_ was presented at the `LSST 2016 Project and Community Workshop <https://project.lsst.org/meetings/lsst2016/>`_.
+This document supercedes that content.
+
+Enabling Python 3
+=================
+
+At the time of writing Python 3 porters must use an `lsstsw <https://developer.lsst.io/build-ci/lsstsw.html>`_ environment to do the development.
+A new environment must be created and should not be shared with a Python 2 stack installation.
+In particular, do not share an EUPS database across both stacks.
+This is because the Python binary interfaces are not compatible between 2 and 3, and also EUPS currently has a bug resulting in database corruption when Python 2 attempts to read a database that was updated with Python 3.
+
+.. note ::
+    When sourcing the ``lsstsw`` ``setup.sh`` script, check that no EUPS environment variables
+    have been defined. EUPS will not overwrite these variables when doing the setup, leading to
+    possible confusion (and data corruption) over which EUPS database is active.
+
```

Comment:
You might want to change `note` to `caution` or `warning`. Sometimes a reader moving quickly through the page might skip notes but is more likely to read a caution or warning, and this is a pretty important point.